src/libraries/ConstantProductMath.sol

```diff
@@ -4,6 +4,12 @@ pragma solidity ^0.8.0;
 library ConstantProductMath {
     uint256 public constant FEE_ACCURACY = 1_000_000; // 100%
 
+    /// @dev the function assumes that the following args are within the respective bounds as enforced by ReservoirPair
+    /// and therefore would not overflow
+    /// aAmountIn   <= uint104
+    /// aReserveIn  <= uint104
+    /// aReserveOut <= uint104
+    /// aSwapFee    <= 0.2e6
     function getAmountOut(uint256 aAmountIn, uint256 aReserveIn, uint256 aReserveOut, uint256 aSwapFee)
```

Comment:
You should mention that breaching these args is safe (it will revert) rather than invalid behaviour

Reply:
fixed